platform/saivpp/vpplib/SwitchStateBaseAcl.cpp

```diff
@@ -597,65 +898,100 @@ sai_status_t SwitchStateBase::AclTblConfig(
 				meta->attridname);
 	    }
 	    
-	    if (attr->id == SAI_ACL_ENTRY_ATTR_FIELD_ACL_RANGE_TYPE) {
+        if(attr->id == SAI_ACL_ENTRY_ATTR_FIELD_TUNNEL_TERMINATED) {
+            if(attr->value.aclfield.data.booldata == true) {
+                update_tunterm_rule = true;
+            }
+        }
+
+	    if (attr->id == SAI_ACL_ENTRY_ATTR_FIELD_ACL_RANGE_TYPE && acl->count>0  && (rule<(&acl->rules[0]+n_entries))) {
 		for (uint32_t jdx = 0; jdx < p_ace->range_count; jdx++) {
 		    status = acl_rule_port_range_vpp_acl_set(p_ace->range_type[jdx],
 							     &p_ace->range_limit[jdx], rule);
 		}
 	    } else {
-		status = acl_rule_field_update((sai_acl_entry_attr_t) attr->id, &attr->value, rule);
-	    }
+            if(acl->count>0 && (rule<(&acl->rules[0]+n_entries))) {
+		        status = acl_rule_field_update((sai_acl_entry_attr_t) attr->id, &attr->value, rule);
+            }
+            if(tunterm_acl->count>0 && (tunterm_rule<(&tunterm_acl->rules[0]+n_tunterm_entries))) {
+                status = tunterm_acl_rule_field_update((sai_acl_entry_attr_t) attr->id, &attr->value, tunterm_rule);
+	        }
+        }
 	}
-	rule++;
+    
+    if(update_tunterm_rule) {
+        if(acl->count>0 && (rule<(&acl->rules[0]+n_entries))) {
+            memset(rule, 0, sizeof(*rule)); // clear any fields that were added to regular rule
+        }
+        tunterm_rule++;
+    } else {
+        if(tunterm_acl->count>0 && (tunterm_rule<(&tunterm_acl->rules[0]+n_tunterm_entries))) {
+            memset(tunterm_rule, 0, sizeof(*tunterm_rule)); // clear any fields that were added to tunterm_rule
+	    }
+        rule++;
+    } 
     }
 
     bool acl_replace;
     uint32_t acl_swindex;
-
     auto vpp_idx_it = m_acl_swindex_map.find(tbl_oid);
-    if (vpp_idx_it == m_acl_swindex_map.end()) {
-	acl_swindex = 0;
-	acl_replace = false;
-    } else {
-	acl_swindex = vpp_idx_it->second;
-	acl_replace = true;
+    if(acl->count > 0) {
```

Comment:
This block shouldn't have changed, yet it's marked as a diff here due to indentation changes. Please revert this.

Reply:
The check for acl->count > 0 was added here to skip this block for the cases where there are only tunterm acl rules and no standard acl rules